### PR TITLE
Merge PostExtra in to Post

### PR DESCRIPTION
**Pull request by symroe**
Here we go again.

Not a lot to report here, this is more or less the same as the other PRs that merge extra models.

It was a little complex moving the PostExtraElection M2M, just because Django isn't great at knowing how to do this. I've tested this on a DB dump with a load of data in and it's worked each time.

As per normal, it's hard to know about false negatives here, so testing by clicking about will be helpful.

A also added a couple of new tests for some uses of PostExtra that weren't picked up by the existing tests. I pulled the tests in to a new commit, but the code changes got bundled up in the main commit.

### CODE REVIEW

**Comment by chris48s** [comment]
This isn't a full review, but in the interests of "failing fast", I'm now getting every "post" page (e.g: `/election/local.city-of-london.2018-08-30/post/LBW:E05009293/bread-street` ) throwing `unorderable types: NoneType() < int()` in `constituencies.py` with the code on this branch.

Does that reproduce for you?

**Comment by symroe** [comment]
On the `constituencies.py` bug – I think this is because of this change:

https://github.com/DemocracyClub/yournextrepresentative/pull/621/files#diff-0be2f1a571e5b7e2f1ce2416dc0a272bR904

It's actually a little unrelated to the work to merge the two models, so I could pick it out. The reason I did it was because the `get_max_winners` function was failing due to some other reason (I don't remember) and I thought that we don't want to look at the `people_elected_per_post` property of `Election` if we can help it.

Basically, we should set `1`, not `NULL` as the default for `PostExtraElection.winner_count`. 

What do you reckon – pick it out and fix whatever failed before, or include it but add a migration to set the default for `PostExtraElection`?

Or to ask a slightly different question – is a winner count of None _ever_ useful?

File: ynr/apps/popolo/migrations/0012_move_post_extra_data_to_base.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.8 on 2018-08-21 20:44
```
[the remark this answers is not in the document]
Comment (symroe) [reply ynr/apps/popolo/migrations/0012_move_post_extra_data_to_base.py]: I thin the Django version and time are helpful (although I guess we can infer both from git) – but I can write "Generated by Django 1.10.8 on 2018-08-21 20:44, edited by Sym Roe" or something?

File: ynr/apps/popolo/migrations/0009_move_extra_person_data_to_base.py [hunk 2]
```diff
@@ -28,6 +29,11 @@ def add_extra_fields_to_base(apps, schema_editor):
     Image.objects.filter(content_type_id=pe_content_type_id).update(
         content_type_id=p_content_type_id
     )
+    images = Image.objects.filter(content_type_id=p_content_type_id)
+    for image in images:
+        if image.object_id in old_to_new_map:
+            image.object_id = old_to_new_map[image.object_id]
+            image.save()
```
[the remark this answers is not in the document]
Comment (symroe) [reply ynr/apps/popolo/migrations/0009_move_extra_person_data_to_base.py]: Yep, I've picked this out in to a new commit now – wasn't meant to be included here

**Comment by symroe** [comment]
Right, that should be everything addresses?

**Comment by symroe** [comment]
Alright, a couple more commits pushed to deal with the max_winners issue. It's very much moving in to refactoring logic and not just merging models, but I think in the context it's fine.

**Comment by chris48s** [comment]
Agree with the thrust of the solution in 9c50868 :+1:  but it looks like there are now some test failures to fix

**Comment by symroe** [comment]
Oh _who_ added Black to the CI 🙄

**Comment by symroe** [comment]
🎉 Almost there now…
